Elastic agent: Add logging content (#808)
elastic/observability-docs: docs/en/ingest-management/elastic-agent/elastic-agent-standalone-logging.asciidoc

[[elastic-agent-standalone-logging-config]]
[role="xpack"]

> = Configure logging for {agent} standalone

Reviewer: ```suggestion
= Configure logging for standalone {agent}s
```

Reviewer: I'd say either "standalone Elastic Agents" or "Elastic Agents running standalone"

Author: Got it. Thanks!

The Logging section of the `elastic-agent.yml` config file contains settings for configuring the logging output.

> The logging system can write logs to the `syslog`, `file`, `stderr`, `eventlog` or rotate log files.

Reviewer: ```suggestion
The logging system can write logs to the `syslog`, `file`, `stderr`, `eventlog`, or rotate log files.
```

Reviewer: oxford comma FTW

Author: Ah!

publishing, set to `publish`. Multiple selectors can be chained.

Possible values: `[beat]`, `[publish]`, `[service]`

| `agent.logging.to_stderr` | Set to `true` to write all logging output to the `stderr` output—this is equivalent to
using the `-e` command line option.

Default: `true`

| `agent.logging.to_syslog` | Set to `true` to write all logging output to the `syslog` output.

Default: `false`

| `agent.logging.to_eventlog` | Set to `true` to write all logging output to the Windows `eventlog` output.

Default: `false`

| `agent.logging.metrics.enabled` | Set to `true` for {agent} to periodically log its internal metrics that have changed in the
last period. For each metric that changed, the delta from the value at the beginning of the period is logged.
Also, the total values for all non-zero internal metrics get logged on shutdown.

Default: `true`

| `agent.logging.metrics.period` | Specify the period after which to log the internal metrics.

Default: `30s`

| `agent.logging.to_files` | Set to `true` to log to rotating files. Set to `false` to disable logging to files.

Default: `true`

| `agent.logging.files.path` | The directory that log files is written to.

> include::{tab-widgets}/logging-widget.asciidoc[]

Reviewer: cool...wasn't sure if a widget would work inside of a table cell. It would be interesting to see what the experience is like someone using a screen reader. (Probably not worth your time to test this right now because at some point we'll do some accessibility testing that will flag issues.)

Reviewer 2: Very neat

The value must be in the range of `2` to `1024` files.

Default: `7`

| `agent.logging.files.permissions` | The permissions mask to apply when rotating log files. The permissions option
must be a valid Unix-style file permissions mask expressed in octal notation. In Go, numbers in octal notation must start with 0.

Default: `0600`

| `agent.logging.files.interval` | Enable log file rotation on time intervals in addition to the size-based rotation. Intervals must be at least `1s`.
Values of `1m`, `1h`, `24h`, `7*24h`, `30*24h`, and `365*24h` are boundary-aligned with minutes, hours, days, weeks, months, and years as
reported by the local system clock. All other intervals get calculated from the Unix epoch.

Default: `0` (disabled)

| `agent.logging.files.rotateonstartup` | Set to `true` to rotate existing logs on startup rather than to append to the existing file. 

Default: `true`

| `agent.logging.json` | Set to `true` to log messages in JSON format. 

Default: `false`

| `agent.logging.ecs` | Set to `true` to log messages with the minimal required Elastic Common Schema (ECS)

> information. We recommended to use this option in combination with `agent.logging.json: true`.

Reviewer 2: ```suggestion
information. We recommended using this option in combination with `agent.logging.json: true`.
```
